packages/codecs-data-structures/src/data-enum.ts

```diff
@@ -239,13 +240,13 @@ function getDataEnumMaxSize<const TVariants extends Variants<Decoder<any> | Enco
 
 function getVariantDiscriminator<const TVariants extends Variants<Decoder<any> | Encoder<any>>>(
     variants: TVariants,
-    discriminatorValue: string,
+    discriminatorValue: DiscriminatorValue,
 ) {
     const discriminator = variants.findIndex(([key]) => discriminatorValue === key);
     if (discriminator < 0) {
         throw new SolanaError(SOLANA_ERROR__CODECS__INVALID_DATA_ENUM_VARIANT, {
-            value: discriminatorValue,
-            variants: variants.map(([key]) => key),
+            value: String(discriminatorValue),
+            variants: variants.map(([key]) => String(key)),
```

Comment:
Mmmm… one of the powerful things about error context is that your _program_ can access the actual, original values. Here we're kneecapping your `catch` block, in that it can't see what the original value was – only what its string representation is.

I think the message formatter does `String(…)` already; is that enough?

Reply:
Good shout. I wasn't sure if the error context was gonna be happy with that but I'm glad to hear it can handle it. I'll make the change.